otherlibs/stdune/bytes_unit.ml

```diff
@@ -0,0 +1,37 @@
+let conversion_table =
+  [ ([ "bytes"; "B" ], 1L)
+  ; ([ "KB"; "kB"; "KiB"; "kilobytes" ], 1000L)
+  ; ([ "MB"; "mb"; "MiB"; "megabytes" ], 1000_000L)
+  ; ([ "GB"; "gB"; "GiB"; "gigabytes" ], 1000_000_000L)
+  ]
+
+let pp x =
+  (* We go through the list to find the first unit that is greater than the
+     number of bytes and take the predecessor as the units for printing. For the
+     special base case where no conversion is necessery we don't print as a
+     float. *)
+  let suffix, value =
+    let rec loop = function
+      | [] -> assert false
+      | [ (units, value) ] -> (List.hd units, value)
+      | (_, value) :: ((units', value') :: _ as l) ->
+        if value >= x && x > value' then (List.hd units', value') else loop l
+    in
+    loop @@ List.rev conversion_table
+  in
+  if value = 1L then Printf.sprintf "%Ld %s" x suffix
+  else
+    Printf.sprintf "%.2f %s" (Int64.to_float x /. Int64.to_float value) suffix
+
+let () =
```

Comment:
It seems weird to test by running code at the top level. Can we turn this into a proper test that runs only as part of the test suite, not on every execution of Dune?

Reply:
I've moved it into an expect test. Not sure what I was thinking when I wrote it.